lib/libutils/isoc/bget_malloc.c

```diff
@@ -765,9 +765,8 @@ static void gen_malloc_add_pool(struct malloc_ctx *ctx, void *buf, size_t len)
 	uint32_t exceptions;
 	uintptr_t start = (uintptr_t)buf;
 	uintptr_t end = start + len;
-	const size_t min_len = ((sizeof(struct malloc_pool) + (SizeQuant - 1)) &
-					(~(SizeQuant - 1))) +
-				sizeof(struct bhead) * 2;
+	const size_t min_len = ROUNDUP(sizeof(struct malloc_pool), SizeQuant) +
```

Comment:
I guess this part isn't needed any longer, that is, we don't care about `sizeof(struct malloc_pool)` if I get it right.

Reply:
IIUC a pool must at least fit the size of `struct bhead` and `struct bfhead`. 
Keep at least `min_len = sizeof(struct bhead) + sizeof(struct bfhead) + SizeQuant`?

Reply:
Isn't `min_len = sizeof(struct bhead) + sizeof(struct bfhead)` enough?

Reply:
Yes, that should be enough.
One `struct bhead` is always used as a sentinel at the end of this buffer.
A free block is represented with a `struct bfhead`, while a used block only uses a `struct bhead`. So there should be room for at least `struct qlinks` or `2 * sizeof(void *)`.
That's ridiculously small, but if we reject that where should we draw the limit instead?

Reply:
Ok, makes sense to me. Thanks a lot for the help.